src/OAuth/AuthorizationServiceInterface.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace ZfrLightspeedRetail\OAuth;
+
+use Psr\Http\Message\UriInterface;
+use ZfrLightspeedRetail\Exception\InvalidStateException;
+use ZfrLightspeedRetail\Exception\MissingRequiredScopeException;
+use ZfrLightspeedRetail\Exception\UnauthorizedException;
+
+/**
+ * @author Daniel Gimenes
+ */
+interface AuthorizationServiceInterface
+{
+    /**
+     * Builds an authorization URL that identifies the internal account ID and the requested scope.
+     *
+     * @param string   $referenceId    Internal account ID (identifies the account in your application)
```

Comment:
Okay, got it by reading the rest of the code. It's used to distinguish nonce until you get the account ID.